SwiftUIDZ/SwiftUIDZ/ClimateControlView/ClimateControlView.swift

```diff
@@ -0,0 +1,350 @@
+//
+//  ClimateControlView.swift
+//  SwiftUIDZ
+//
+//  Created by Алена Панченко on 31.01.2023.
+//
+
+import SwiftUI
+
+/// Вью - модель настроек системы кондиционирования
+final class ClimateControlViewModel: ObservableObject {
+    
+    @Published var gradientColors = [Color("GradientBackgroundTop"), Color("GradientBackGroundBottom")]
+    @Published var settingsIndex = 0
+ 
+   
+    @Published var settingsClimate = SettingsClimateModel.getSettings()
+    
+    @Published var climateSliderValues = [0.0, 0.0, 0.0, 0.0]
+    
+    @Published var isClimateControlEnabled = false
+    @Published var bottomSheetCurrentOffsetY: CGFloat = 0
+    @Published var bottomSheetLastOffsetY: CGFloat = 0
+    @Published var bottomGradientColor: CGColor = CGColor(red: 47, green: 184, blue: 255, alpha: 1)
+    
+  
+    @Published var topGradientColor: CGColor = CGColor(red: 158, green: 236, blue: 217, alpha: 1)
+    
+    // MARK: - Public Methods
+    
+//    func getSettingsName() -> String {
+//        settingsClimate[settingsIndex].settingName
+//    }
+//
+//    func getSettingsImageName() -> String {
+//        settingsClimate[settingsIndex].settingsImageName
+//    }
+//
+//    func getSettingsSliderMaxValue() -> Double {
+//        settingsClimate[settingsIndex].sliderMaxValue
+//    }
+//
+//    func toggleActionSettings(condition: Bool) {
+//        settingsClimate[settingsIndex].isAction = condition
+//    }
+    
+    func isActive() {
+        if !settingsClimate[settingsIndex].isAction {
+            climateSliderValues[settingsIndex] = 0
+        }
+    }
+}
+/// Настройки системы кондиционирования
+struct ClimateControlView: View {
+
+    // MARK: - Private Constants
+    
+    private enum Constants {
+        static let gradientTopColorName = "GradientTop"
+        static let gradientBottomColorName = "GradientBottom"
+    }
+    
+   
+    
+ // MARK: - Public Properties
+    
+    @State var isSlider = 0.0
+    @State var isrange = 0.0...15
+    
+    @Environment(\.presentationMode) var presentation
+    @GestureState private var bottomSheetGestureOffsetValue = CGSize.zero
+    @StateObject private var climateControlViewModel = ClimateControlViewModel()
+  
+    var dragBottomSheetGesture: some Gesture {
+        DragGesture()
+            .updating($bottomSheetGestureOffsetValue) { value, state, _ in
+                state = value.translation
+                onChangeBottomSheetOffset()
+            }
+            .onEnded { _ in
+                withAnimation {
```

Comment:
```suggestion
             withAnimation {
                     if -climateControlViewModel.bottomSheetCurrentOffsetY > 200 {
                         climateControlViewModel.bottomSheetCurrentOffsetY = -100
                     } else {
                         climateControlViewModel.bottomSheetCurrentOffsetY = 0
                     }
                     climateControlViewModel.bottomSheetLastOffsetY = climateControlViewModel.bottomSheetCurrentOffsetY
```
в vm вынеси это же логика